dtc-lexer.l

```diff
@@ -151,6 +151,19 @@ static void PRINTF(1, 2) lexical_error(const char *fmt, ...);
 			return DT_LABEL;
 		}
 
+<V1>{LABEL} 	{
+			/* Missed includes or macro definitions while
+			 * preprocessing can lead to unexpected identifiers in
+			 * the input. Report a slightly more informative error
+			 * in this case */
+
+			lexical_error("Unexpected '%s'", yytext);
+
+			/* Guess that it was supposed to be a macro that expands
+			 * to a literal for purposes of further processing */
+			return DT_LITERAL;
```

Comment:
This isn't right as is.  You're returning DT_LITERAL - which represents an integer literal - to the parser, but you're not populating `yylval.integer`, which means this will have some leftover, possibly uninitialized value.  Even though it's likely to error out, shortly after this, allowing non-deterministic behaviour here is not a good idea.

You could set `yylval.integer = 0`, essentially treating the unknown identifier as zero.  Alternatively you could return a new `DT_IDENTIFIER` token type.  That would force a syntax error in the parser level, since nothing would recognize it.

Reply:
I have added initialization of `yylval.integer`.

I've found `DT_LITERAL` generates more helpful error messages, so I'm keeping that part.